### PR TITLE
audit-log: Add auditconfigupdater to machine manifolds

### PR DESCRIPTION
**Pull request by babbageclunk**
## Description of change

The auditconfigupdater was added to the 2.3 branch but wasn't running in the 2.4 branch because the API server has been moved into the dependency engine. This adds a manifold for the auditconfigupdater, adds the worker to the machine agent manifolds, and updates the API server worker and manifold to receive the GetAuditConfig func as an input.

## QA steps

Bootstrap a new controller with auditing enabled. Check that messages are written to the audit log. Turn off auditing `juju controller-config auditing-enabled=false`. See that the update is mentioned in the debug log, and that messages for new commands aren't written to the audit log. Re-enable auditing and see that new commands are added again. Check that upgrading juju works, and that changing the controller config can be done afterwards.

### CODE REVIEW

File: worker/auditconfigupdater/manifold.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	logDir := agent.CurrentConfig().LogDir()
+
+	st := statePool.SystemState()
+
+	logFactory := func(cfg auditlog.Config) auditlog.AuditLog {
+		return auditlog.NewLogFile(logDir, cfg.MaxSizeMB, cfg.MaxBackups)
+	}
+	md.auditConfig, err = initialConfig(st)
+	if err != nil {
+		stTracker.Done()
```
Comment (wallyworld) [inline worker/auditconfigupdater/manifold.go]: we should instead add a defer to the top of this function - remembering to call Done() everywhere is quite error prone. of course, this requires a named err return argument
defer func() {
  if err != nil && stTracker != nil {
    stTracker.Done()
  }
}

File: worker/auditconfigupdater/manifold_test.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+	statetesting "github.com/juju/juju/state/testing"
+	coretesting "github.com/juju/juju/testing"
+	"github.com/juju/juju/worker/auditconfigupdater"
+	"github.com/juju/juju/worker/dependency"
+	dt "github.com/juju/juju/worker/dependency/testing"
+	"github.com/juju/juju/worker/workertest"
+)
+
+type manifoldSuite struct {
+	statetesting.StateSuite
```
Comment (wallyworld) [inline worker/auditconfigupdater/manifold_test.go]: no new worker tests should use StateSuite
or is it too hard to mock everything?

File: worker/auditconfigupdater/manifold.go (hunk 1 above)
Comment (babbageclunk) [reply worker/auditconfigupdater/manifold.go]: :ok:

File: worker/auditconfigupdater/manifold_test.go (hunk 2 above)
Comment (babbageclunk) [reply worker/auditconfigupdater/manifold_test.go]: I don't think it'll be hard to mock it out - I'll do that.

Comment (babbageclunk) [reply worker/auditconfigupdater/manifold_test.go]: Hmm actually, since the state tracker returns a `*state.StatePool` it's really awkward to mock out.

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by jameinel** [comment]
Seems like this broke the actual dependency engine. Maybe the new worker is failing to start and is blocking everything else?
52209f81-0fe4-417d-83e1-3a692fc20694: waiting for [action-pruner agent api-caller api-config-watcher application-scaler charm-revision-updater clock compute-provisioner environ-tracker firewaller instance-poller is-responsible-flag log-forwarder machine-undertaker metric-worker migration-fortress migration-inactive-flag migration-master model-upgrade-gate model-upgraded-flag not-alive-flag not-dead-flag remote-relations state-cleaner status-history-pruner storage-provisioner unit-assigner]
52209f81-0fe4-417d-83e1-3a692fc20694: unexpected []
52209f81-0fe4-417d-83e1-3a692fc20694: report: 
{}

**Comment by babbageclunk** [comment]
Ugh, chasing this down now. :(

**Comment by babbageclunk** [comment]
I haven't managed to reprodue this locally - running the suite under the stress script doesn't get it to fail.

(The changed commit building now is just a rebase.)

**Comment by babbageclunk** [comment]
I'm putting this down to an intermittent failure that I probably didn't cause?

$$merge$$

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
$$merge$$

**Pull request merged**